content/blog/2022/dependabot-single-pull-request-python.md

```diff
@@ -0,0 +1,132 @@
+[requested](https://github.com/dependabot/dependabot-core/issues/1190) since
+[2018](https://github.com/dependabot/dependabot-core/issues/2265). As of May
+2022, the feature is on [Dependabot's
+roadmap](https://github.com/github/roadmap/issues/148), but doesn't have a
+concrete release date.
+
+At OddBird we decided to leverage [GitHub
+Actions](https://github.com/features/actions) to produce a pull request with all
+Python dependencies updated at once. We have created a workflow that runs every
+Monday in the early morning. By the time we start our work day the pull request
+has been opened and we can merge it right away if all tests pass, or debug any
+issues introduced by upstream breaking changes. We also gave us the option of
+[running the workflow on
+demand](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow)
+to upgrade dependencies at our convenience.
+
+Broadly speaking the workflow will:
+
+1. Checkout our code
+1. Configure a Python environment that matches our project version
+1. Run a custom command to upgrade all dependencies
+1. Use git to detect changed files
+1. Commit the changes to a pre-defined branch (if they exist)
+1. Open a pull request for that branch (if it doesn't exist already)
+
+This workflow should be compatible with most Python projects. You might need to
+adjust step 3 to run your own command depending on what package manager you use
+(we use and recommend [pip-tools](https://github.com/jazzband/pip-tools)).
+
+Here's what the resulting [automated pull
+request](https://github.com/oddbird/Metecho/pull/2161) looks like. In this case
+we also included JavaScript changes, but you can see the branch was merged
+cleanly without any extra commits by maintainers. In other cases we had to do
+some clean up to account for breaking changes in our dependencies.
+
+We've been using this approach on several projects now and are pretty satisfied
+with the reduction in PR noise without compromising on security and features.
+So, without further ado, here's the sweet YAML file you can drop into your repo:
+
+```yml
+# File: .github/workflows/upgrade-dependencies.yml
+name: Upgrade dependencies
+
+on:
+  workflow_dispatch: # Allow running on-demand
+  schedule:
+    # Runs every Monday at 8:00 UTC (4:00 Eastern)
+    - cron: '0 8 * * 1'
+
+jobs:
+  upgrade:
+    name: Upgrade & Open Pull Request
+    runs-on: ubuntu-latest
+    env:
+      # This branch will receive updates each time the workflow runs
+      # It doesn't matter if it's deleted when merged, it'll be re-created
+      BRANCH_NAME: auto-dependency-upgrades
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # [Optional] Use a separate key to automatically execute checks on the resulting PR
+          # https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs
+          ssh-key: ${{ secrets.DEPLOY_KEY }}
```

Comment:
These don't show up for me in the compiled version. Maybe we need to escape something here? (Same goes for all `{{ ... }}` in this example.)

![Screen Shot 2022-05-29 at 6 00 39 PM](https://user-images.githubusercontent.com/552316/170899683-f4ed798e-458c-498f-bb05-2d8b7465d020.jpg)

Reply:
Needed a `{% raw %}` tag in there. Fixed in ca808745d9a3beee28d818533412576114d58eed